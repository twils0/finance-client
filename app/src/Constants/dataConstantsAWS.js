// eslint-disable-next-line
export const actionTypes = {
  SET_AWS_STATUS: 'DATA_AUTH_AWS_STATUS',
  SET_AWS_USER: 'DATA_AUTH_SET_AWS_USER',
  RESET_AWS_STATE: 'DATA_AUTH_RESET_AWS_STATE',
};
