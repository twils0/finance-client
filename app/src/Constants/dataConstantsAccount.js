export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
