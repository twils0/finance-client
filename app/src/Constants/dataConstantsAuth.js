export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
