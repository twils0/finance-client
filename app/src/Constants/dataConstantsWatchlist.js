export const actionTypes = {
  SET_WATCHLIST_STATUS: 'DATA_WATCHLIST_SECURITIES_STATUS',
  SET_SECURITIES_CURRENT: 'DATA_WATCHLIST_SET_SECURITIES_CURRENT',
  SET_SECURITIES_LIST: 'DATA_WATCHLIST_SET_SECURITIES_LIST',
  SET_SECURITIES_ALL: 'DATA_WATCHLIST_SET_SECURITIES_ALL',
  SET_SECURITY: 'DATA_WATCHLIST_SET_SECURITY',
  SET_SECURITY_DATA: 'DATA_WATCHLIST_SET_SECURITY_DATA',
  SET_SECURITY_SUBSCRIBE: 'DATA_WATCHLIST_SET_SECURITY_SUBSCRIBE',
  RESET_WATCHLIST_STATE: 'DATA_ACCOUNT_RESET_WATCHLIST_STATE',
};

export const codeTypeNames = {
  CONFIRM_PHONE: 'confirmPhone',
  CONFIRM_EMAIL: 'confirmEmail',
};

export const statusNames = {
  SECURITIES: 'watchlistStatusSecurities',
  GET_SECURITY_DATA: 'watchlistStatusGetSecurityData',
  UPDATE_SECURITIES: 'watchlistStatusUpdateSecurities',
};
