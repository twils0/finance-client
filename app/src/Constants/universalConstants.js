import stripeConfig from '../../../stripe.config.json';
import dbConfig from '../../../db.config.json';
import iexConfig from '../../../iex.config.json';

export const baseURLs = {
  STRIPE: stripeConfig.url,
  DB: dbConfig.url,
  IEX: iexConfig.urls.api,
  IEX_DEV: iexConfig.urls.dev,
  IEX_TERMS: iexConfig.urls.terms,
};

export const URLs = {
  CUSTOMERS: '/customers',
  USERS: '/users',
  EMAILS: '/emails',
  SECURITIES: '/securities',
  SECURITIES_SEARCH: '/securities/search',
  IEX_COMPANY: tickerCusip => `/stock/${tickerCusip}/company`,
  IEX_QUOTE: tickerCusip => `/stock/${tickerCusip}/quote`,
};

export const axiosConfig = {
  STRIPE: {
    baseURL: baseURLs.STRIPE,
    timeout: 30000,
  },
  DB: {
    baseURL: baseURLs.DB,
    timeout: 30000,
  },
  IEX: {
    baseURL: baseURLs.IEX,
    timeout: 30000,
  },
};

export const pathNames = {
  DEFAULT: '*',
  LOGIN: '/login',
  SIGN_UP: '/signup',
  CODE: '/code',
  CODE_VERIFY_EMAIL: '/code/:verificationId?',
  TERMS: '/terms',
  INTERNAL: '/internal',
  WATCHLIST: '/internal/watchlist',
  WATCHLIST_SECURITY_ID: '/internal/watchlist/:securityId?',
  ACCOUNT: '/internal/account',
};

export const requestStatusTypes = {
  IDLE: 'IDLE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
