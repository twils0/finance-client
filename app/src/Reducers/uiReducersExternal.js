import { combineReducers } from 'redux';

import {
  formNames,
  inputNames,
  buttonNames,
  buttonTexts,
  actionTypes,
} from '../Constants/uiConstantsExternal';

export const initialState = {
  utilities: {
    forms: {
      springSettings: { stiffness: 170, damping: 26 },
      width: 300,
      widthBuffer: 60,
      mediaWidth: 700,
      heightBuffer: 50,
      inputs: {
        heightBuffer: 15,
      },
    },
    buttons: {
      width: 300,
      heightBuffer: 20,
      heightBottomBuffer: 30,
    },
    terms: {
      bufferPage: 30,
      bufferHeightParagraphTop: 20,
      bufferHeightParagraphBottom: 30,
      bufferHeight: 20,
    },
  },
  forms: {
    current: formNames.LOGIN,
    list: [...Object.values(formNames)],
    [formNames.RESET_PASSWORD]: {
      id: formNames.RESET_PASSWORD,
      height: 339,
      inputsList: [...Object.values(inputNames[formNames.RESET_PASSWORD])],
      inputs: {
        [inputNames[formNames.RESET_PASSWORD].CODE]: {
          parentId: formNames.RESET_PASSWORD,
          id: inputNames[formNames.RESET_PASSWORD].CODE,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.RESET_PASSWORD].PASSWORD]: {
          parentId: formNames.RESET_PASSWORD,
          id: inputNames[formNames.RESET_PASSWORD].PASSWORD,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.RESET_PASSWORD].PASSWORD2]: {
          parentId: formNames.RESET_PASSWORD,
          id: inputNames[formNames.RESET_PASSWORD].PASSWORD2,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.FORGOT_PASSWORD]: {
      id: formNames.FORGOT_PASSWORD,
      height: 220,
      inputsList: [...Object.values(inputNames[formNames.FORGOT_PASSWORD])],
      inputs: {
        [inputNames[formNames.FORGOT_PASSWORD].EMAIL]: {
          parentId: formNames.FORGOT_PASSWORD,
          id: inputNames[formNames.FORGOT_PASSWORD].EMAIL,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.LOGIN]: {
      id: formNames.LOGIN,
      height: 222,
      inputsList: [...Object.values(inputNames[formNames.LOGIN])],
      inputs: {
        [inputNames[formNames.LOGIN].EMAIL]: {
          parentId: formNames.LOGIN,
          id: inputNames[formNames.LOGIN].EMAIL,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.LOGIN].PASSWORD]: {
          parentId: formNames.LOGIN,
          id: inputNames[formNames.LOGIN].PASSWORD,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.CARD]: {
      id: formNames.CARD,
      height: 360,
      inputsList: [...Object.values(inputNames[formNames.CARD])],
      inputs: {
        [inputNames[formNames.CARD].NAME_ON_CARD]: {
          parentId: formNames.CARD,
          id: inputNames[formNames.CARD].NAME_ON_CARD,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.CARD].PROMO_CODE]: {
          parentId: formNames.CARD,
          id: inputNames[formNames.CARD].PROMO_CODE,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.CARD].STRIPE]: {
          parentId: formNames.CARD,
          id: inputNames[formNames.CARD].STRIPE,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.INFO]: {
      id: formNames.INFO,
      height: 280,
      inputsList: [...Object.values(inputNames[formNames.INFO])],
      inputs: {
        [inputNames[formNames.INFO].NAME]: {
          parentId: formNames.INFO,
          id: inputNames[formNames.INFO].NAME,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.INFO].EMAIL]: {
          parentId: formNames.INFO,
          id: inputNames[formNames.INFO].EMAIL,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.INFO].PHONE]: {
          parentId: formNames.INFO,
          id: inputNames[formNames.INFO].PHONE,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.SIGN_UP]: {
      id: formNames.SIGN_UP,
      height: 247,
      inputsList: [...Object.values(inputNames[formNames.SIGN_UP])],
      inputs: {
        [inputNames[formNames.SIGN_UP].PASSWORD]: {
          parentId: formNames.SIGN_UP,
          id: inputNames[formNames.SIGN_UP].PASSWORD,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.SIGN_UP].PASSWORD2]: {
          parentId: formNames.SIGN_UP,
          id: inputNames[formNames.SIGN_UP].PASSWORD2,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.CODE_MFA_PHONE]: {
      id: formNames.CODE_MFA_PHONE,
      height: 183,
      inputsList: [...Object.values(inputNames[formNames.CODE_MFA_PHONE])],
      inputs: {
        [inputNames[formNames.CODE_MFA_PHONE].CODE]: {
          parentId: formNames.CODE_MFA_PHONE,
          id: inputNames[formNames.CODE_MFA_PHONE].CODE,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.CODE_VERIFY_PHONE]: {
      id: formNames.CODE_VERIFY_PHONE,
      height: 183,
      inputsList: [...Object.values(inputNames[formNames.CODE_VERIFY_PHONE])],
      inputs: {
        [inputNames[formNames.CODE_VERIFY_PHONE].CODE]: {
          parentId: formNames.CODE_VERIFY_PHONE,
          id: inputNames[formNames.CODE_VERIFY_PHONE].CODE,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.CODE_VERIFY_EMAIL]: {
      id: formNames.CODE_VERIFY_EMAIL,
      height: 183,
      inputsList: [...Object.values(inputNames[formNames.CODE_VERIFY_EMAIL])],
      inputs: {
        [inputNames[formNames.CODE_VERIFY_EMAIL].CODE]: {
          parentId: formNames.CODE_VERIFY_EMAIL,
          id: inputNames[formNames.CODE_VERIFY_EMAIL].CODE,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.DEVICE]: {
      id: formNames.DEVICE,
      height: 146,
      inputsList: [],
      inputs: {},
    },
  },
  buttons: {
    list: [...Object.values(buttonNames)],
    [buttonNames.FIRST]: { id: buttonNames.FIRST, text: buttonTexts.LOG_IN },
    [buttonNames.SECOND]: { id: buttonNames.SECOND, text: buttonTexts.SIGN_UP },
    [buttonNames.FORGOT_PASSWORD]: {
      id: buttonNames.FORGOT_PASSWORD,
      text: buttonTexts.FORGOT_PASSWORD,
    },
    [buttonNames.RESEND_PHONE]: { id: buttonNames.RESEND_PHONE, text: buttonTexts.RESEND_PHONE },
    [buttonNames.RESEND_EMAIL]: { id: buttonNames.RESEND_EMAIL, text: buttonTexts.RESEND_EMAIL },
  },
};

export const utilities = (state = initialState.utilities) => state;

export const forms = (state = initialState.forms, action) => {
  switch (action.type) {
    case actionTypes.SET_CURRENT_FORM:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.SET_FORM_HEIGHT:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.SET_INPUT_VALUE_ERROR:
      return {
        ...state,
        [action.payload.parentId]: {
          ...state[action.payload.parentId],
          inputs: {
            ...state[action.payload.parentId].inputs,
            [action.payload.id]: {
              ...state[action.payload.parentId].inputs[action.payload.id],
              ...action.payload,
            },
          },
        },
      };
    default:
      return state;
  }
};

export const buttons = (state = initialState.buttons, action) => {
  switch (action.type) {
    case actionTypes.SET_BUTTON_TEXT:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    default:
      return state;
  }
};

const uiReducersExternal = combineReducers({
  utilities,
  forms,
  buttons,
});

export default uiReducersExternal;
