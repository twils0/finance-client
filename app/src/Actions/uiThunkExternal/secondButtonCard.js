import { pathNames } from '../../Constants/universalConstants';
import { formNames, inputNames } from '../../Constants/uiConstantsExternal';
import { setInputValueError } from '../uiActionsExternal';

const secondButtonCard = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'history')) {
    throw new Error(`Please enter a value for the 'history' key - ${JSON.stringify(payload)}`);
  }
  if (!Object.prototype.hasOwnProperty.call(payload, 'resetStripeElement')) {
    throw new Error(`Please enter a value for the 'resetStripeElement' key - ${JSON.stringify(payload)}`);
  }

  return (dispatch, getState) => {
    const state = getState();
    const { forms } = state.ui.external;
    const { history } = payload;

    const cardNameOnCard = forms[formNames.CARD].inputs[inputNames[formNames.CARD].NAME_ON_CARD];
    const cardPromoCode = forms[formNames.CARD].inputs[inputNames[formNames.CARD].PROMO_CODE];
    const cardStripe = forms[formNames.CARD].inputs[inputNames[formNames.CARD].STRIPE];

    const infoName = forms[formNames.INFO].inputs[inputNames[formNames.INFO].NAME];
    const infoEmail = forms[formNames.INFO].inputs[inputNames[formNames.INFO].EMAIL];
    const infoPhone = forms[formNames.INFO].inputs[inputNames[formNames.INFO].PHONE];

    const signUpPassword = forms[formNames.SIGN_UP].inputs[inputNames[formNames.SIGN_UP].PASSWORD];
    const signUpPassword2 =
      forms[formNames.SIGN_UP].inputs[inputNames[formNames.SIGN_UP].PASSWORD2];

    history.replace(pathNames.LOGIN);

    if (cardNameOnCard.value || cardNameOnCard.errorMessage) {
      cardNameOnCard.value = '';
      cardNameOnCard.errorMessage = '';

      dispatch(setInputValueError(cardNameOnCard));
    }

    if (cardPromoCode.value || cardPromoCode.errorMessage) {
      cardPromoCode.value = '';
      cardPromoCode.errorMessage = '';

      dispatch(setInputValueError(cardPromoCode));
    }

    if (cardStripe.value || cardStripe.errorMessage) {
      cardStripe.value = '';
      cardStripe.errorMessage = '';

      dispatch(setInputValueError(cardStripe));
      payload.resetStripeElement();
    }

    if (infoName.value || infoName.errorMessage) {
      infoName.value = '';
      infoName.errorMessage = '';

      dispatch(setInputValueError(infoName));
    }

    if (infoEmail.value || infoEmail.errorMessage) {
      infoEmail.value = '';
      infoEmail.errorMessage = '';

      dispatch(setInputValueError(infoEmail));
    }

    if (infoPhone.value || infoPhone.errorMessage) {
      infoPhone.value = '';
      infoPhone.errorMessage = '';

      dispatch(setInputValueError(infoPhone));
    }

    if (signUpPassword.value || signUpPassword.errorMessage) {
      signUpPassword.value = '';
      signUpPassword.errorMessage = '';

      dispatch(setInputValueError(signUpPassword));
    }

    if (signUpPassword2.value || signUpPassword2.errorMessage) {
      signUpPassword2.value = '';
      signUpPassword2.errorMessage = '';

      dispatch(setInputValueError(signUpPassword2));
    }

    return null;
  };
};

export default secondButtonCard;
