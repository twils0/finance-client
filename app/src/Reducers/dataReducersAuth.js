import { combineReducers } from 'redux';

import { requestStatusTypes, pathNames } from '../Constants/universalConstants';
import { actionTypes, statusNames, codeTypeNames } from '../Constants/dataConstantsAuth';

export const initialState = {
  authenticated: false,
  redirectURL: pathNames.WATCHLIST,
  status: {
    [statusNames.LOGIN]: {
      id: statusNames.LOGIN,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_OUT_DEVICES]: {
      id: statusNames.SIGN_OUT_DEVICES,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGIN_MFA]: {
      id: statusNames.LOGIN_MFA,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_UP]: {
      id: statusNames.SIGN_UP,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE]: {
      id: statusNames.VERIFY_PHONE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE_CODE]: {
      id: statusNames.VERIFY_PHONE_CODE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL]: {
      id: statusNames.VERIFY_EMAIL,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL_LINK]: {
      id: statusNames.VERIFY_EMAIL_LINK,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.FORGOT_PASSWORD]: {
      id: statusNames.FORGOT_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.RESET_PASSWORD]: {
      id: statusNames.RESET_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.CHANGE_PASSWORD]: {
      id: statusNames.CHANGE_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGOUT]: {
      id: statusNames.LOGOUT,
      status: requestStatusTypes.SUCCESS,
    },
    [statusNames.DELETE_ACCOUNT]: {
      id: statusNames.DELETE_ACCOUNT,
      status: requestStatusTypes.SUCCESS,
    },
  },
  codeTypes: {
    [codeTypeNames.VERIFY_PHONE]: {
      id: codeTypeNames.VERIFY_PHONE,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL]: {
      id: codeTypeNames.VERIFY_EMAIL,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL_ADDITIONAL]: {
      id: codeTypeNames.VERIFY_EMAIL_ADDITIONAL,
      needed: true,
    },
  },
};

export const authenticated = (state = initialState.authenticated, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTHENTICATED:
      return action.payload.authenticated;
    case actionTypes.RESET_AUTH_STATE:
      return initialState.authenticated;
    default:
      return state;
  }
};

export const redirectURL = (state = initialState.redirectURL, action) => {
  switch (action.type) {
    case actionTypes.SET_REDIRECT_URL:
      return action.payload.redirectURL;
    default:
      return state;
  }
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTH_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const codeTypes = (state = initialState.codeTypes, action) => {
  switch (action.type) {
    case actionTypes.SET_CODE_TYPE:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.codeTypes;
    default:
      return state;
  }
};

const dataReducersAuth = combineReducers({
  authenticated,
  redirectURL,
  status,
  codeTypes,
});

export default dataReducersAuth;
