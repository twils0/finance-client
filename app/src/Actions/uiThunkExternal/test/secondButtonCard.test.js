import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { initialState as initialStateAuth } from '../../../Reducers/dataReducersAuth';
import { initialState as initialStateUIExternal } from '../../../Reducers/uiReducersExternal';
import { pathNames } from '../../../Constants/universalConstants';
import {
  actionTypes as actionTypesUIExternal,
  formNames,
  inputNames,
} from '../../../Constants/uiConstantsExternal';
import secondButtonCard from '../secondButtonCard';

const history = {
  replace: jest.fn(),
};

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const cardPayload = {
  history,
  resetStripeElement: jest.fn(),
};

const nameOnCard = 'testNameOnCard';
const promoCode = 'testPromoCode';
const name = 'testName';
const email = 'test@test.com';
const phone = '239-555-0000';
const password = 'testPassword1!';

describe('uiThunkAccount', () => {
  describe('secondButtonCard', () => {
    afterEach(() => {
      history.replace.mockReset();
      cardPayload.resetStripeElement.mockReset();
    });

    it("fails and throws an error when missing 'history' key in payload", async () => {
      const store = mockStore({
        data: { auth: initialStateAuth },
        ui: { external: initialStateUIExternal },
      });

      const emptyPayload = {};
      const expectedActions = [];

      try {
        await store.dispatch(secondButtonCard(emptyPayload));
      } catch (errorCatch) {
        expect(errorCatch).toMatchSnapshot();
      }

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });

    it("fails and throws an error when missing 'resetStripeElement' key in payload", async () => {
      const store = mockStore({
        data: { auth: initialStateAuth },
        ui: { external: initialStateUIExternal },
      });

      const wrongPayload = {
        history,
      };
      const expectedActions = [];

      try {
        await store.dispatch(secondButtonCard(wrongPayload));
      } catch (errorCatch) {
        expect(errorCatch).toMatchSnapshot();
      }

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });

    it('creates the correct actions with the correct payload, inputs all have values', async () => {
      const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));
      const { forms } = stateBeforeUIExternal;

      forms[formNames.CARD].inputs[inputNames[formNames.CARD].NAME_ON_CARD].value = nameOnCard;
      forms[formNames.CARD].inputs[inputNames[formNames.CARD].STRIPE].value = 'complete';
      forms[formNames.CARD].inputs[inputNames[formNames.CARD].PROMO_CODE].value = promoCode;
      forms[formNames.INFO].inputs[inputNames[formNames.INFO].NAME].value = name;
      forms[formNames.INFO].inputs[inputNames[formNames.INFO].EMAIL].value = email;
      forms[formNames.INFO].inputs[inputNames[formNames.INFO].PHONE].value = phone;
      forms[formNames.SIGN_UP].inputs[inputNames[formNames.SIGN_UP].PASSWORD].value = password;
      forms[formNames.SIGN_UP].inputs[inputNames[formNames.SIGN_UP].PASSWORD2].value = password;

      const cardNameOnCard = forms[formNames.CARD].inputs[inputNames[formNames.CARD].NAME_ON_CARD];
      const cardPromoCode = forms[formNames.CARD].inputs[inputNames[formNames.CARD].PROMO_CODE];
      const cardStripe = forms[formNames.CARD].inputs[inputNames[formNames.CARD].STRIPE];

      const infoName = forms[formNames.INFO].inputs[inputNames[formNames.INFO].NAME];
      const infoEmail = forms[formNames.INFO].inputs[inputNames[formNames.INFO].EMAIL];
      const infoPhone = forms[formNames.INFO].inputs[inputNames[formNames.INFO].PHONE];

      const signUpPassword =
        forms[formNames.SIGN_UP].inputs[inputNames[formNames.SIGN_UP].PASSWORD];
      const signUpPassword2 =
        forms[formNames.SIGN_UP].inputs[inputNames[formNames.SIGN_UP].PASSWORD2];

      const payloadCardNameOnCard = JSON.parse(JSON.stringify(cardNameOnCard));
      const payloadCardPromoCode = JSON.parse(JSON.stringify(cardPromoCode));
      const payloadCardStripe = JSON.parse(JSON.stringify(cardStripe));
      const payloadInfoName = JSON.parse(JSON.stringify(infoName));
      const payloadInfoEmail = JSON.parse(JSON.stringify(infoEmail));
      const payloadInfoPhone = JSON.parse(JSON.stringify(infoPhone));
      const payloadSignUpPassword = JSON.parse(JSON.stringify(signUpPassword));
      const payloadSignUpPassword2 = JSON.parse(JSON.stringify(signUpPassword2));

      payloadCardNameOnCard.value = '';
      payloadCardNameOnCard.errorMessage = '';

      payloadCardPromoCode.value = '';
      payloadCardPromoCode.errorMessage = '';

      payloadCardStripe.value = '';
      payloadCardStripe.errorMessage = '';

      payloadInfoName.value = '';
      payloadInfoName.errorMessage = '';

      payloadInfoEmail.value = '';
      payloadInfoEmail.errorMessage = '';

      payloadInfoPhone.value = '';
      payloadInfoPhone.errorMessage = '';

      payloadSignUpPassword.value = '';
      payloadSignUpPassword.errorMessage = '';

      payloadSignUpPassword2.value = '';
      payloadSignUpPassword2.errorMessage = '';

      const store = mockStore({
        ui: { external: stateBeforeUIExternal },
      });

      const expectedActions = [
        {
          type: actionTypesUIExternal.SET_INPUT_VALUE_ERROR,
          payload: payloadCardNameOnCard,
        },
        {
          type: actionTypesUIExternal.SET_INPUT_VALUE_ERROR,
          payload: payloadCardPromoCode,
        },
        {
          type: actionTypesUIExternal.SET_INPUT_VALUE_ERROR,
          payload: payloadCardStripe,
        },
        {
          type: actionTypesUIExternal.SET_INPUT_VALUE_ERROR,
          payload: payloadInfoName,
        },
        {
          type: actionTypesUIExternal.SET_INPUT_VALUE_ERROR,
          payload: payloadInfoEmail,
        },
        {
          type: actionTypesUIExternal.SET_INPUT_VALUE_ERROR,
          payload: payloadInfoPhone,
        },
        {
          type: actionTypesUIExternal.SET_INPUT_VALUE_ERROR,
          payload: payloadSignUpPassword,
        },
        {
          type: actionTypesUIExternal.SET_INPUT_VALUE_ERROR,
          payload: payloadSignUpPassword2,
        },
      ];

      const result = await store.dispatch(secondButtonCard(cardPayload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(result).toEqual(null);
      expect(history.replace).toBeCalledWith(pathNames.LOGIN);
      expect(cardPayload.resetStripeElement).toBeCalled();
    });

    it('creates the correct actions with the correct payload, inputs all are blank', async () => {
      const store = mockStore({
        ui: { external: initialStateUIExternal },
      });

      const expectedActions = [];

      const result = await store.dispatch(secondButtonCard(cardPayload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(result).toEqual(null);
      expect(history.replace).toBeCalledWith(pathNames.LOGIN);
      expect(cardPayload.resetStripeElement).not.toBeCalled();
    });
  });
});
