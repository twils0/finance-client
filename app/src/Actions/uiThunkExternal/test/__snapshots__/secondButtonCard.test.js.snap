// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uiThunkExternal secondButtonCard fails and throws an error when missing 'history' key in payload 1`] = `[Error: Please enter a value for the 'history' key - {}]`;

exports[`uiThunkExternal secondButtonCard fails and throws an error when missing 'resetStripeElement' key in payload 1`] = `[Error: Please enter a value for the 'resetStripeElement' key - {"history":{}}]`;
